pkg/store/bucket.go

```diff
@@ -1585,11 +1632,22 @@ func (r *bucketChunkReader) addPreload(id uint64) error {
 }
 
 // preload all added chunk IDs. Must be called before the first call to Chunk is made.
-func (r *bucketChunkReader) preload() error {
+func (r *bucketChunkReader) preload(samplesLimiter *Limiter) error {
 	const maxChunkSize = 16000
 
 	var g run.Group
 
+	numChunks := uint64(0)
+	for _, offsets := range r.preloads {
+		for range offsets {
+			numChunks++
+		}
+	}
+	numSamples := numChunks * maxSamplesPerChunk
```

Comment:
let's inline this maybe?

Reply:
Will do, no need to create an extra variable.